core/LSU.cpp

```diff
@@ -87,48 +91,78 @@ namespace olympia
     // Send initial credits (ldst_inst_queue_size_) to Dispatch Unit
     void LSU::sendInitialCredits_()
     {
+        // Setup scoreboard view upon register file
+        std::vector<core_types::RegFile> reg_files = {core_types::RF_INTEGER, core_types::RF_FLOAT};
+        for(const auto rf : reg_files)
+        {
+            scoreboard_views_[rf].reset(new sparta::ScoreboardView(getContainer()->getName(),
+                                                                   core_types::regfile_names[rf],
+                                                                   getContainer()));
+        }
         out_lsu_credits_.send(ldst_inst_queue_size_);
 
         ILOG("LSU initial credits for Dispatch Unit: " << ldst_inst_queue_size_);
     }
 
     // Receive new load/store instruction from Dispatch Unit
     void LSU::getInstsFromDispatch_(const InstPtr & inst_ptr)
-    {
-        // Create load/store memory access info
-        MemoryAccessInfoPtr mem_info_ptr = sparta::allocate_sparta_shared_pointer<MemoryAccessInfo>(memory_access_allocator,
-                                                                                                    inst_ptr);
-
-        // Create load/store instruction issue info
-        LoadStoreInstInfoPtr inst_info_ptr = sparta::allocate_sparta_shared_pointer<LoadStoreInstInfo>(load_store_info_allocator,
-                                                                                                       mem_info_ptr);
-        lsu_insts_dispatched_++;
- 
-        // Append to instruction issue queue
-        appendIssueQueue_(inst_info_ptr);
-
-        // Update issue priority & Schedule an instruction issue event
-        updateIssuePriorityAfterNewDispatch_(inst_ptr);
-        uev_issue_inst_.schedule(sparta::Clock::Cycle(0));
+    {   
+        core_types::RegFile reg_file = core_types::RF_INTEGER;
+        const auto & srcs = inst_ptr->getSourceOpInfoList();
+        const auto & dests = inst_ptr->getDestOpInfoList();
+        if(dests.size() > 0){
+            sparta_assert(dests.size() == 1); // we should only have one destination
+            reg_file = determineRegisterFile(dests[0]);
+        }
```

Comment:
Why are you using the destination register as the register file type?

Reply:
I used the destination register for the register file type because we have to mark the destination register as ready for loads in the LSU in the corresponding scoreboard.